scripts/menus/network.lua

```diff
@@ -512,187 +516,205 @@ function RunServerMultiGameMenu(map, description, numplayers, options)
       end
     end
     if startIn < -1 then
-       startIn = startIn + 1
+      startIn = startIn + 1
     else
-       if optDedicated then
-	  dedicated:setMarked(true)
-	  dedicatedCb(dedicated)
-	  optDedicated = false
-       elseif (optRace == "human" or optRace == "Human") then
-	  race:setSelected(1)
-	  raceCb(race)
-	  optRace = ""
-       elseif (optRace == "orc" or optRace == "Orc") then
-	  race:setSelected(2)
-	  raceCb(race)
-	  optRace = ""
-       elseif (options.fow == 0) then
-	  fow:setMarked(false)
-	  fowCb(fow)
-	  options.fow = -1
-       elseif (options.revealmap == 1) then
-	  revealmap:setMarked(true)
-	  revealMapCb(revealmap)
-	  options.revealmap = -1
-       elseif (optAutostartNum) then
-	  if (optAutostartNum <= readyplayers) then
-	     if (startIn < 0) then
-		startIn = 100
-	     else
-		startIn = startIn - 1
-		if (startIn == 0) then
-		   startFunc()
-		end
-	     end
-	     waitingtext:setCaption("Starting in " .. startIn / 2)
-	     print("Starting in " .. startIn / 2)
-	  end
-       else
-	  startgame:setVisible(ready)
-	  waitingtext:setVisible(not ready)
-       end
+      if optDedicated then
+        dedicated:setMarked(true)
+        dedicatedCb(dedicated)
+        optDedicated = false
+
+      elseif (optRace == "human" or optRace == "Human") then
+        race:setSelected(1)
+        raceCb(race)
+        optRace = ""
+      elseif (optRace == "orc" or optRace == "Orc") then
+        race:setSelected(2)
+        raceCb(race)
+        optRace = ""
+
+      elseif (optResources == "low" or optResources == "Low") then
+        resources:setSelected(1)
+        resourcesCb(resources)
+        optResources = ""
+      elseif (optResources == "medium" or optResources == "Medium") then
+        resources:setSelected(2)
+        resourcesCb(resources)
+        optResources = ""
+      elseif (optResources == "high" or optResources == "High") then
+        resources:setSelected(3)
+        resourcesCb(resources)
+        optResources = ""
+
+      elseif (optUnits == "1") then
+        units:setSelected(1)
+        unitsCb(resources)
+        optUnits= ""
+
+      elseif (options.fow == 0) then
+        fow:setMarked(false)
+        fowCb(fow)
+        options.fow = -1
+      elseif (options.revealmap == 1) then
+        revealmap:setMarked(true)
+        revealMapCb(revealmap)
+        options.revealmap = -1
+      elseif (optAutostartNum) then
+        if (optAutostartNum <= readyplayers) then
+          if (startIn < 0) then
+            startIn = 100
+          else
+            startIn = startIn - 1
+            if (startIn == 0) then
+              startFunc() end
```

Comment:
You mean 'startFunc() end' should be split in two lines, right? I agree
with you. It's probably that I was trying to manually indent the code
and saw these 'func() end' elsewere in lua code. Then I used some
external formatter.

Besides the formatting the code works for me. Don't you think these long
functions should be split into more but shorter functions? I could try
to make that.

-- 
Karol Kreński

> > +      elseif (options.fow == 0) then
> > +        fow:setMarked(false)
> > +        fowCb(fow)
> > +        options.fow = -1
> > +      elseif (options.revealmap == 1) then
> > +        revealmap:setMarked(true)
> > +        revealMapCb(revealmap)
> > +        options.revealmap = -1
> > +      elseif (optAutostartNum) then
> > +        if (optAutostartNum <= readyplayers) then
> > +          if (startIn < 0) then
> > +            startIn = 100
> > +          else
> > +            startIn = startIn - 1
> > +            if (startIn == 0) then
> > +              startFunc() end
> 
> This doesn't look right ... Shouldn't it be
> ```lua
> if (startIn < 0) then
>   startIn = 100
> else
>   startIn = startIn  - 1
>   if (startIn == 0) then
>     startFunc()
>   end
> end
> ```
> ?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/Wargus/wargus/pull/157/files#r53552786